pkg/api/customization/cluster/validator.go

```diff
@@ -270,3 +275,72 @@ func (v *Validator) accessTemplate(request *types.APIContext, spec *mgmtclient.C
 
 	return nil
 }
+
+// validateGenericEngineConfig allows for additional validation of clusters that depend on Kontainer Engine or Rancher Machine driver
+func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec *v3.ClusterSpec) error {
+
+	if request.Method == http.MethodPost {
+		return nil
+	}
+
+	clusterName := request.ID
+	prevCluster, err := v.ClusterLister.Get("", clusterName)
+	if err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, err.Error())
+	}
+	if spec.AmazonElasticContainerServiceConfig != nil {
+		err := validateEKS(*prevCluster.Spec.GenericEngineConfig, *spec.AmazonElasticContainerServiceConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func validateEKS(prevCluster, newCluster map[string]interface{}) error {
+	// check config is for EKS clusters
+	if driver, ok := prevCluster["driverName"]; ok {
+		if driver != service.AmazonElasticContainerServiceDriverName {
+			return nil
+		}
+	}
+
+	// don't allow for updating subnets
+	if prev, ok := prevCluster["subnets"]; ok {
+		if new, ok := newCluster["subnets"]; ok {
+			prevSlice, ok := prev.([]interface{})
+			if !ok {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "unable to convert subnets to interace slice")
+			}
+			newSlice, ok := new.([]interface{})
+			if !ok {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "unable to convert subnets to interace slice")
+			}
+
+			if !testEq(prevSlice, newSlice) {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "cannot modify EKS subnets after creation")
+			}
+		}
+	}
+	return nil
+}
+
+func testEq(a, b []interface{}) bool {
```

Comment:
Why are we rewriting `DeepEqual`?

Reply:
Avoids the use of reflection when we know the type

Reply:
Switched to use `reflect.DeepEqual`